examples/beamforming/mvdr.py

```diff
@@ -0,0 +1,418 @@
+"""Implementation of MVDR Beamforming Module
+
+Based on https://github.com/espnet/espnet/blob/master/espnet2/enh/layers/beamformer.py
+
+We provide three solutions of MVDR beamforming. One is based on reference channel selection:
+Souden, Mehrez, Jacob Benesty, and Sofiene Affes.
+"On optimal frequency-domain multichannel linear filtering for noise reduction."
+IEEE Transactions on audio, speech, and language processing 18.2 (2009): 260-276.
+
+The other two solutions are based on the steering vector. We apply either eigenvalue decomposition
+or the power method to get the steering vector from the PSD matrices.
+
+For eigenvalue decomposistion method, please refer:
+Higuchi, Takuya, et al. "Robust MVDR beamforming using time-frequency masks for online/offline ASR in noise."
+2016 IEEE International Conference on Acoustics, Speech and Signal Processing (ICASSP). IEEE, 2016.
+
+For power method, please refer:
+Mises, R. V., and Hilda Pollaczek‐Geiringer.
+"Praktische Verfahren der Gleichungsauflösung."
+ZAMM‐Journal of Applied Mathematics and Mechanics/Zeitschrift für Angewandte Mathematik und Mechanik 9.1 (1929): 58-77.
+
+For online streaming audio, we provide a recursive method to update PSD matrices based on:
+Higuchi, Takuya, et al.
+"Online MVDR beamformer based on complex Gaussian mixture model with spatial prior for noise robust ASR."
+IEEE/ACM Transactions on Audio, Speech, and Language Processing 25.4 (2017): 780-793.
+"""
+
+from typing import Optional
+import torch
+
+
+def mat_trace(input: torch.Tensor) -> torch.Tensor:
+    r"""Compute the trace of a Tensor
+
+    Args:
+        input (torch.Tensor): Tensor of dimension (..., channel, channel)
+
+    Returns:
+        torch.Tensor: trace of the input Tensor
+    """
+    assert input.shape[-1] == input.shape[-2]
+    input = torch.diagonal(input, 0, dim1=-1, dim2=-2)
+    return input.sum(dim=-1)
+
+
+class PSD(torch.nn.Module):
+    r"""Compute cross-channel power spectral density (PSD) matrix.
+
+    Args:
+        normalize (bool): whether normalize the mask over the time dimension
```

Comment:
nit: `(bool, optional)`

the torchaudio convention for docstrings uses `optional` to refer there being a default value (and not the `Optional` type). For instance, `(T, optional)` if for a variable of type T with a default value, and `(T or None, optional)` for a variable of type `Optional[T]` with a default value. I see some other inconsistencies with this in this PR